### PR TITLE
[libpq] Update mingw patch

### LINKED ISSUE #24911: [libpq] Build error
**Host Environment**
- OS: [Windows11 x64]
- Compiler: mingw 8.1.0 64bit

**To Reproduce**
`./vcpkg install gdal  --triplet=x64-mingw-dynamic`

**Failure logs**
-Cut and paste the appropriate build messages from the console output.
-Please attach any additional failure logs mentioned in the console output.
```cmd
G:\buildtmp\vcpkg>vcpkg.exe install gdal  --triplet=x64-mingw-dynamic
Computing installation plan...
The following packages will be built and installed:
    gdal[core,hdf5,netcdf,postgresql,supported-default-features]:x64-mingw-dynamic -> 3.4.3
  * libiconv[core]:x64-mingw-dynamic -> 1.16#13
  * libpq[core,lz4,openssl,zlib]:x64-mingw-dynamic -> 14.1#1
  * libwebp[core,libwebpmux,nearlossless,simd,unicode]:x64-mingw-dynamic -> 1.2.2
  * libxml2[core]:x64-mingw-dynamic -> 2.9.14
  * netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-mingw-dynamic -> 4.8.1#2
  * openjpeg[core]:x64-mingw-dynamic -> 2.5.0
  * pkgconf[core]:x64-windows -> 1.8.0#3
  * vcpkg-pkgconfig-get-modules[core]:x64-windows -> 2022-02-10
  * vcpkg-tool-meson[core]:x64-windows -> 0.60.2#2
  * zstd[core]:x64-mingw-dynamic -> 1.5.2#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-mingw-dynamic...
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\o\AppData\Local\vcpkg\archives in 390.9 us. Use --debug to see more details.
Installing 1/11 libpq:x64-mingw-dynamic...
Building libpq[core,lz4,openssl,zlib]:x64-mingw-dynamic...
-- Using community triplet x64-mingw-dynamic. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: G:\buildtmp\vcpkg\triplets\community\x64-mingw-dynamic.cmake
-- Using cached postgresql-14.1.tar.bz2.
-- Extracting source G:/buildtmp/vcpkg/downloads/postgresql-14.1.tar.bz2
-- Applying patch patches/windows/install.patch
-- Applying patch patches/windows/win_bison_flex.patch
-- Applying patch patches/windows/openssl_exe_path.patch
-- Applying patch patches/windows/Solution.patch
-- Applying patch patches/windows/MSBuildProject_fix_gendef_perl.patch
-- Applying patch patches/windows/msgfmt.patch
-- Applying patch patches/windows/python_lib.patch
-- Applying patch patches/windows/fix-compile-flag-Zi.patch
-- Applying patch patches/windows/tcl_version.patch
-- Applying patch patches/fix-configure.patch
-- Applying patch patches/mingw/link-with-crypt32.patch
CMake Error at scripts/cmake/z_vcpkg_apply_patches.cmake:64 (message):
  Applying patch failed: Checking patch configure...

  warning: configure has type 100644, expected 100755

  error: while searching for:



  fi





  if test "$with_readline" = yes; then





  error: patch failed: configure:11708

  error: configure: patch does not apply

  Checking patch configure.in...

  error: configure.in: No such file or directory

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive.cmake:237 (z_vcpkg_apply_patches)
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:32 (vcpkg_extract_source_archive)
  ports/libpq/portfile.cmake:76 (vcpkg_extract_source_archive_ex)
  scripts/ports.cmake:146 (include)


error: building libpq:x64-mingw-dynamic failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libpq
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libpq]+Build+error
Include '[libpq] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-05-05-67e17c1782801cf481be9ac0b3765dff3e4bdeb8
    vcpkg-scripts version: 93b38546e 2022-05-24 (5 hours ago)
```

**Additional context**
eidt  `ports/libpq/portfile.cmake`,commented `55` line.
```
if(VCPKG_TARGET_IS_MINGW)
    #list(APPEND PATCHES patches/mingw/link-with-crypt32.patch)
endif()
```
compilation passed.

### PR DESCRIPTION
**Pull request by LilyWangLL**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #24911, fix mingw patch applly failed

### CODE REVIEW

File: ports/libpq/patches/mingw/link-with-crypt32.patch [hunk 1]
```diff
@@ -62,18 +62,19 @@ index 75210f0..6ea5e1f 100755
...
+diff --git a/configure.ac b/configure.ac
+index 286213f..badf3c3 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -1173,7 +1173,7 @@ AC_SEARCH_LIBS(clock_gettime, [rt posix4])
+ # Solaris:
+ AC_SEARCH_LIBS(fdatasync, [rt posix4])
  # Cygwin:
- AC_SEARCH_LIBS(shmget, cygipc)
+-AC_SEARCH_LIBS(shmget, cygipc)
```
Comment (JackBoosY) [inline ports/libpq/patches/mingw/link-with-crypt32.patch:76]: Why remove  this line?

**Comment by dg0yt** [comment]
1. I think the mingw patch should be transformed so that it can always be applied, including vpkg CI. That's the only way to avoid new breakage.
2. The mingw patch touches `configure` but this is pointless: `configure` is regenerated during `vcpkg_configure_make`. Remove that generated part to make the patch more stable.
3. Maybe the patch can be dropped by always passing `LIBS=crypt32` with `OPTIONS`, for mingw.

**Comment by LilyWangLL** [comment]
> 1. I think the mingw patch should be transformed so that it can always be applied, including vpkg CI. That's the only way to avoid new breakage.
> 2. The mingw patch touches `configure` but this is pointless: `configure` is regenerated during `vcpkg_configure_make`. Remove that generated part to make the patch more stable.
> 3. Maybe the patch can be dropped by always passing `LIBS=crypt32` with `OPTIONS`, for mingw.

Thanks for your suggestion, I change this PR to draft, and I first correct my mistake. Could you submit your suggestion about mingw? I don't have the mingw machine. We could also ask @sotex's help, ask him help test.

**Comment by dg0yt** [comment]
See #24941 for a more thorough approach, consistently using `pkg-config`, instead of stumbling from one issue to the next.

**Comment by LilyWangLL** [comment]
Close this PR due to replaced by #24941.

**Pull request closed without merging**